client/js/widgets/widget.js

```diff
@@ -2302,6 +2312,18 @@ export class Widget extends StateManaged {
       await this.set('rotation', degrees);
   }
 
+  async selected() {
+    if (!this.addOverlay || window.getComputedStyle(addOverlay).display === 'none')
+      this.set('selectedBy', playerName);
+      this.domElement.classList.add('selected');
```

Comment:
This doesn't work the way you want it to. `this.domElement.classList` will be overridden destructively whenever something about the widget changes.

See `setHighlighted` which basically does the same thing for selecting widgets in the editor. It sets a flag on the object itself (`this`) and then that flag is used whenever `this.domElement.classList` is updated.